power_grid_model_c/power_grid_model/include/power_grid_model/main_model_impl.hpp

```diff
@@ -714,50 +766,113 @@ class MainModelImpl<ExtraRetrievableTypes<ExtraRetrievableType...>, ComponentLis
   public:
     template <class Component> using UpdateType = typename Component::UpdateType;
 
-    static bool is_update_independent(ConstDataset const& update_data) {
-        // If the batch size is (0 or) 1, then the update data for this component is 'independent'
-        if (update_data.batch_size() <= 1) {
-            return true;
-        }
+    static UpdateCompIndependence check_components_independence(ConstDataset const& update_data) {
+        auto check_ids_na = [](auto const& all_spans) {
+            std::vector<std::vector<bool>> ids_na{};
+            for (const auto& span : all_spans) {
+                for (const auto& obj : span) {
+                    std::vector<bool> id_na{};
+                    if constexpr (requires { obj.id; }) {
+                        id_na.push_back(obj.id == na_Idx);
+                    }
+                    ids_na.push_back(id_na);
+                }
+            }
+            return ids_na;
+        };
```

Comment:
this also looks like something that can go to `update.hpp`

Reply:
Refactor scoped out. Noted in https://github.com/PowerGridModel/power-grid-model/issues/806